code/modules/mob/living/living_defines.dm

```diff
@@ -139,3 +139,6 @@
 
 	/// This is the cooldown on suffering additional effects for when we exhaust all stamina
 	COOLDOWN_DECLARE(last_stamina_exhaustion)
+
+	var/lastgundelay //This is the time after which the next gun can be fired in world time.
+	var/lastgun //This is a reference to the last gun that the mob fired.
```

Comment:
This is extremly messy. I assume you need that to deal with akimbo, but you could probably just use dual_weild var

Reply:
Its for preventing attachment juggling AND prevents you from just holding an ammoless gun to bypass dual_wield.

It also makes it so that low fire rate guns don't affect the fire rate of their attachments.

The other PR uses a different solution.